src/metabase/sync/sync_metadata/fks.clj

```diff
@@ -93,4 +77,4 @@
           {:total-fks    0
            :updated-fks  0
            :total-failed 0}
-          (sync-util/db->sync-tables database)))
+          (sync-util/db->reducible-sync-tables database)))
```

Comment:
This fixes an issue where we hold all tables in memory during this sync step. This is okay for most databases but could become problematic when there are 100,000s of tables.